corehq/apps/data_dictionary/util.py

```diff
@@ -360,3 +364,9 @@ def is_case_type_deprecated(domain, case_type):
         return case_type_obj.is_deprecated
     except CaseType.DoesNotExist:
         return False
+
+
+def is_case_type_or_prop_name_valid(case_prop_name):
```

Comment:
I'm not sure where we currently do validation when creating a property/case type through the UI, but should we not use that rather?

Reply:
You can have a look [here](https://github.com/dimagi/commcare-hq/pull/33740) for how we will do the UI validation (PR still needs to be merged).

Given that the UI validation is done on the front-end through JavaScript and the import validation is being handled on the back-end through Django, we unfortunately wouldn't be able to easily use what we've done in the other PR. 

We could do the reverse of the above however, and instead move the front-end validation to the back-end to make use of the util function I've created, but I feel that refactor might be better suited for a separate PR.

Reply:
Agreed, thanks for the clarification!